### ISSUE TITLE
Incrementally enable new Pylint lints

### ISSUE THREAD

**Issue by Eric-Arellano** [Qiskit/qiskit-terra#9614]
Our previous Pylint version was 2.8, and I recently upgraded it to 2.16 in .... To facilitate landing it, I disabled a bunch of checks. But most of them would be good to activate.

Working on this PR means:

1. Choose one of the checks you want to enable from here:
2. Delete it from `.pylintrc`
3. Run `tox -e lint` and fix all the issues until `tox -e lint` succeeds.

To make it easier for code reviewers, it's a good idea to activate 1 or 2 checks at a time.

**Comment by lambda-knight** [comment]
Fixing consider-using-f-string (C0209) is worth trying ?

**Comment by Eric-Arellano** [comment]
Yes, @lambda-knight consider-using-f-string looks useful to me. Note though that we are transitioning to Ruff instead of Pylint. So it may be worth double checking that Ruff has a similar lint. 

Even if Ruff doesn't have it, I still think it's valuable to fix all our historical instances regardless of if we introduce future "regressions" or not. Let's say we have 200 issues now, and we introduce 50 in the next 60 months. Better to have 200/250 fixed than 0/250.

**Comment by lambda-knight** [comment]
OK, @Eric-Arellano ,
Enabling consider-using-f-string, 192 files and 520 warnings are reported.  I am fixing code of this warning and will issue PR.

**Comment by joesho112358** [comment]
@lambda-knight hi, can i help out with `consider-using-f-string`?
